src/__lib__/staticComponent.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+import { Component } from 'preact';
+
+const contentReferences = [];
+
+const staticComponent = (MyComponent, contentReference) =>
```

Comment:
i wonder if this component should get the data `contentReference` describes and pass down to the child? at least that would make it break if that was wrong?

Reply:
then we could just remove the key from the initial props in body too, rather than being a dom ref, and just pull from the dom here if that key is missing?

Reply:
We can certainly pass the data to `staticComponent`, but we'd still need a way to know which keys to omit from `window.gu.app.data.content` in `_html.js`, which is the other thing this module provides a register of those keys, not sure how else we could do this (that doesn't carry the same risks)...

Reply:
OK, I see what you're saying, I did initially open a connection to the content store in `staticComponent` to retrieve the data for these components, which meant there was no way `contentReference` could be wrong as no data would show in the DOM if it were!

I moved it back out after thinking it felt wrong to having a connection in both `Article.js` and `staticComponent.js`.

Reply:
yeah i know what you mean. i think that's what got me to the idea of a`CAPIComponent` – it can explicitly manage the connection, because you're only declaring that it represents part of the CAPI response? 

so it might look something like:

```js
const Headline = styled(CAPIComponent('headline'))({ styles });
```
?

Reply:
where a `CAPIComponent` will always, and only, connect to the the `content` field, and in turn cause that field to be omitted from the initial props in the page body?

Reply:
gonna move to chat...